Service/DatabaseConfigurationService.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace Becklyn\MysqlDumpBundle\Service;
+
+
+use Becklyn\MysqlDumpBundle\Entity\DatabaseConnection;
+use Doctrine\Bundle\DoctrineBundle\Registry;
+use Doctrine\DBAL\Connection;
+use Symfony\Component\DependencyInjection\ContainerAware;
+
+class DatabaseConfigurationService extends ContainerAware
+{
+    /**
+     * @var Registry
+     */
+    private $doctrine;
+
+
+    /**
+     * @var array
+     */
+    private $connectionIdentifiers;
+
+
+    /**
+     * @var string
+     */
+    private $backupPath;
+
+
+    /**
+     * ConfigurationService constructor.
+     *
+     * @param Registry $doctrine
+     * @param array    $connectionIdentifiers
+     * @param string   $backupPath
+     */
+    public function __construct (Registry $doctrine, array $connectionIdentifiers, $backupPath)
+    {
+        $this->doctrine              = $doctrine;
+        $this->connectionIdentifiers = $connectionIdentifiers;
+        $this->backupPath            = !empty($backupPath) ? $backupPath : '';
+    }
+
+
+    /**
+     * Returns all database connections that should be backed up by searching the configuration:
+     *  1) Uses, if provided, the arguments passed via CLI
+     *  2) Searches the config.yml for the key 'becklyn_mysql_dump:databases'
+     *  3) Searches the config.yml for the connections set up under 'doctrine:dbal:connections'
+     *
+     * @param array $cliArguments
+     *
+     * @return array An associative array of identifier => DatabaseConnection
+     */
+    public function getDatabases ($cliArguments)
+    {
+        // If any command line arguments are passed we prefer those ...
+        if (!empty($cliArguments))
+        {
+            return $this->getConnectionDataByIdentifier($cliArguments);
+        }
+
+        // ... alternatively fall back to the previously configured databases ...
+        if (!empty($databases = $this->getConfiguredDatabases()))
+        {
+            return $databases;
+        }
+
+        // ... or fallback to the full list of associated databases configured in Symfony.
+        return $this->getAppDatabases();
+    }
+
+
+    /**
+     * Returns a list of all databases that are configured under 'becklyn_mysql_dump:databases'
+     *
+     * @return array An associative array of identifier => DatabaseConnection
```

Comment:
I once had DatabaseConnection[] instead of the associative array which we're using now until the moment I realised I need some way to list DatabaseConnections I couldn't resolve. If I go for DatabaseConnection[] only I would need to create dummy objects and test whether they could be resolved or not.

The only way we could achieve this is by introducing yet another field to the DatabaseConnection class to flag whether the connection could be resolved. I'm just not sure how clean it is to create a new DatabaseConnection object which doesn't have any values except its identifier and the flag to mark it unresolved/incomplete.

I'm open for suggesstions.